alternator/streams.cc

```diff
@@ -1020,7 +1020,9 @@ future<executor::request_return_type> executor::get_records(client_state& client
         }
 
         // ugh. figure out if we are and end-of-shard
-        return cdc::get_local_streams_timestamp().then([this, iter, high_ts, start_time, ret = std::move(ret)](db_clock::time_point ts) mutable {
+        auto normal_token_owners = _proxy.get_token_metadata_ptr()->count_normal_token_owners();
```

Comment:
So unless I'm missing something, the get_local_streams_timestamp() is no longer used - so shouldn't it be removed?
Aren't we worried about the reduced performance of the GetRecords call? If the generations only change rarely, why do we need to scan the generations table every time? Shouldn't we try to cache the knowledge of the generation?

Reply:
Also, please expand the commit message a bit. You said "Get timestamp via system_distributed, instead of local gen.", but you don't say why. What's wrong with the "local gen" and what is this change fixing?

Reply:
The two should be equal but comes from different tables. (Both are a cql query). Old one from system.local, new one from the actual cdc generation timestamp table. The latter is more in sync with what is actually written into cdc generation data we consume, so is the more correct to use. 
The old is part of generation management, and is used by other parts of that.

Reply:
Basically `get_local_streams_timestamp` is internal API used for internal generation management (it's used for persisting the gossiped generation timestamp so it can be restored on restart).
The `system_distributed.cdc_generation_timestamps` is for clients to access.